### PR TITLE
cheatsheet- adding links to InputFamily

### PR DESCRIPTION
**Pull request by mosheKerbel**
I acceidntly merged and already reverted the origin [PR](https://github.com/wix/wix-style-react/pull/3929) of this 
so this is a new copy of the old PR
@TalZa1810
 remove your local branch `git branch -D cheatsheet/urlsMapping`
pull master `git checkout master && git pull`
checkout this upto date branch `git checkout cheatsheet/urlsMapping`

### CODE REVIEW

File: stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  TimeInput: 'TimeInput',
+  DateInput: 'DateInput',
+  Search: 'Search',
+  ImageViewer: 'ImageViewer',
+};
+
+const symbolsToComponents = [
+  {
+    //3.1
+    UxStory: symbolsNames.Text_Input,
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js]: let's rename this to `symbolName`

File: stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  {
+    //3.15
+    UxStory: symbolsNames.Media_Input,
+    componentsNames: [componentsNames.FormField, componentsNames.ImageViewer],
+  },
+];
+
+export const inputFamilyMapping = {
+  familyName: familyNames.inputs,
+  symbols: symbolsToComponents,
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js]: this mapping is incorrect, inputs should not point to all symbols

File: stories/Introduction/Cheatsheet/ComponentsFamilies/InputFamily.js [hunk 3]
```diff
@@ -24,165 +31,279 @@ import FormField from 'wix-style-react/FormField';
...
+  );
+};
 
-const NumberRangeExample = () => (
-  <SingleComponentSideBySide name="3.5 Number Range Input">
-    <NotDeveloped />
-  </SingleComponentSideBySide>
-);
+const TextAreaExample = () => {
+  const { UxStory, componentsNames } = symbols[1];
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/ComponentsFamilies/InputFamily.js]: So I think it might not be a good idea to treat these symbols as an array.
since you know it's the `Text Area` symbol, why not just pick this for the symbols to components list

File: stories/Introduction/Cheatsheet/ComponentsFamilies/InputFamily.js [hunk 4]
```diff
@@ -24,165 +31,279 @@ import FormField from 'wix-style-react/FormField';
...
+  );
+};
 
-const NumberRangeExample = () => (
-  <SingleComponentSideBySide name="3.5 Number Range Input">
-    <NotDeveloped />
-  </SingleComponentSideBySide>
-);
+const TextAreaExample = () => {
+  const { symbol, components } = symbols[1];
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/ComponentsFamilies/InputFamily.js]: I think it's bad to loop for it in the symbols array.
since you know its a `Text Area`, just pick the relevant symbol

File: stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js (hunk 2 above)
Comment (TalZa1810) [reply stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js]: I did the simboleToComponent mapping only for the Inputs Family. I thought to create a separate mapping for each components family.

File: stories/Introduction/Cheatsheet/ComponentsFamilies/InputFamily.js (hunk 3 above)
Comment (TalZa1810) [reply stories/Introduction/Cheatsheet/ComponentsFamilies/InputFamily.js]: I wanted to iterate the array and add the props dynamically (something similar to the [visual tests](https://github.com/wix/wix-style-react/blob/master/src/IconButton/test/IconButton.visual.js#L7)) meaning creating the `SingleSideBySide` component dynamically but after re-thinking about it, maybe it's not the best solution since there are some edge cases that make this complicated to implement

**Comment by TalZa1810** [comment]
Hi @mosheKerbel ,
I can't add you as a reviewer since you've opened the PR :) 
Can you please have a look whenever?
I've removed the broken links, changed the mapping array to an object and added `onHover` to the story link.

Meanwhile, I've decided **not** to create the `SingleComponentSideBySide` dynamically since there are different scenarios for each case which makes this hard to implement. 


WDYT?

Thanks :)

File: stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js (hunk 2 above)
Comment (shlomitc) [reply stories/Introduction/Cheatsheet/urlsMapping/inputsUrlsMapping.js]: of course. this shouldn't include all symbols, that's just redundant

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  [inputComponentsNames.DateInput]: linkTo('WIP', 'DateInput'),
+  [inputComponentsNames.MultiSelect]: linkTo('Components', 'Multiselect'),
+};
+
+const symbolActualMapping = {
+  [inputSymbols.colorInput]: linkTo(symbolsGroup.inputs, '3.11 ColorInput'),
+  [inputSymbols.tagsInput]: linkTo(symbolsGroup.inputs, '3.12 Tags'),
+};
+
+const componentsWithoutDocumentation = {
```
Comment (TalZa1810) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: I needed this mapping to in case only some of the components are documented. For example, for `ColorInput`the `<FormField/>` is documented but the `<ColorInput/>` is not documented. It's also a good mapping for us to know which components documentation are missing.

**Comment by TalZa1810** [comment]
Hi @shlomitc ,
I've finished with the mappings and files splitting as we discussed. I wasn't sure about some of the files & directory naming.

Let me know what you think.
Thanks :)

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/sketchSymbols.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  tooltipPopovers: '7. Tooltips & Popovers',
+  notificationBars: '8. Notification Bars',
+  modals: '9. Modals',
+  //no links from the sidebar
+  pickers: '10. Pickers',
+  other: '11. Other',
+  contentWidgets: '12. Content Widgets',
+};
+
+export const inputSymbols = {
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/sketchSymbols.js]: `inputsSymbols`

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/InputsFamily.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
+//3. Inputs
+import { inputSymbols } from './sketchSymbols';
+import { inputComponentsNames as componentsNames } from './components';
+
+export const inputsSymbolsToComponents = {
+  //3.1
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/InputsFamily.js]: remove comments

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/components.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+export const inputComponentsNames = {
+  Input: 'Input',
+  Dropdown: 'Dropdown',
+  FormField: 'FormField',
+  InputArea: 'InputArea',
+  InputWithOptions: 'InputWithOptions',
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/components.js]: that's not an input

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/components.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+export const inputComponentsNames = {
+  Input: 'Input',
+  Dropdown: 'Dropdown',
+  FormField: 'FormField',
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/components.js]: should be separated

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/components.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+export const inputComponentsNames = {
+  Input: 'Input',
+  Dropdown: 'Dropdown',
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/components.js]: this one as well

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  });
+
+//Actual mapping
+const componentsActualMapping = {
+  [inputComponentsNames.TimeInput]: linkTo('Components', 'TimePicker'),
+  [inputComponentsNames.DateInput]: linkTo('WIP', 'DateInput'),
+  [inputComponentsNames.MultiSelect]: linkTo('Components', 'Multiselect'),
+};
+
+const symbolActualMapping = {
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: `symobolActualUrl`

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    return {
+      text: componentStr,
+      url: createComponentUrl({ componentName }),
+    };
+  });
+
+//Actual mapping
+const componentsActualMapping = {
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: `componentActualUrl`

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    return componentsActualMapping[componentName];
+  }
+
+  return linkTo('Components', componentName);
+};
+
+export const linkedSymbolName = ({ groupSymbol, symbol }) => {
+  let symbolUrl;
+
+  if (symbolActualMapping[symbol]) {
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: `symbolUrl = symbolActualMapping[symbol] || createSymbolUrl({ groupSymbol, symbol });`

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    symbolUrl = symbolActualMapping[symbol];
+  } else {
+    symbolUrl = createSymbolUrl({ groupSymbol, symbol });
+  }
+
+  return {
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: return only the url

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+const createComponentUrl = ({ componentName }) => {
+  if (componentsActualMapping[componentName]) {
+    return componentsActualMapping[componentName];
+  }
+
+  return linkTo('Components', componentName);
+};
+
+export const linkedSymbolName = ({ groupSymbol, symbol }) => {
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: `getSymbolUrl`

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 16]
```diff
@@ -0,0 +1,59 @@
+import { linkTo } from '@storybook/addon-links';
+import { inputComponentsNames } from './components';
+import { inputSymbols, symbolsGroup } from './sketchSymbols';
+
+const createSymbolUrl = ({ groupSymbol, symbol }) =>
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: can be used inline in the calling function `getSymbolUrl`

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+    symbolUrl = createSymbolUrl({ groupSymbol, symbol });
+  }
+
+  return {
+    text: symbol,
+    url: symbolUrl,
+  };
+};
+
+export const linkedComponentsNames = componentsNamesArr =>
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: same comments as for the above function

File: stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+  componentsNamesArr.map(componentName => {
+    const componentStr = `<${componentName}/>`;
```
Comment (shlomitc) [inline stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: this is the concern of the visual component

Comment (TalZa1810) [reply stories/Introduction/Cheatsheet/symbolsComponentsMapping/storybookMapping.js]: so to resolve this, I thought about adding a `utils` file to the `sharedComponents` directory since these functions, as you mentioned, are more related to the visual shared components.

File: stories/symbolsComponentsMapping/storybookMapping.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+import { inputComponentsNames } from './components';
+import { inputsSymbols, symbolsGroup } from './sketchSymbols';
+
+export const getComponentUrl = ({ componentName }) =>
+  componentActualUrl[componentName] || linkTo('Components', componentName);
+
+export const getSymbolUrl = ({ groupSymbol, symbol }) =>
+  symbolActualUrl[symbol] || linkTo(groupSymbol, symbol);
+
+//Actual mapping
```
Comment (mosheKerbel) [inline stories/symbolsComponentsMapping/storybookMapping.js]: no need for this comment

File: stories/symbolsComponentsMapping/storybookMapping.js [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+const componentActualUrl = {
```
Comment (mosheKerbel) [inline stories/symbolsComponentsMapping/storybookMapping.js]: not sure why 'actual' is required here
just - componentUrl

File: stories/symbolsComponentsMapping/storybookMapping.js [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+  [inputComponentsNames.TimeInput]: linkTo('Components', 'TimePicker'),
+  [inputComponentsNames.DateInput]: linkTo('WIP', 'DateInput'),
+  [inputComponentsNames.MultiSelect]: linkTo('Components', 'Multiselect'),
+};
+
+const symbolActualUrl = {
```
Comment (mosheKerbel) [inline stories/symbolsComponentsMapping/storybookMapping.js]: symbolUrl

File: stories/symbolsComponentsMapping/sketchSymbols.js [hunk 22]
```diff
@@ -0,0 +1,32 @@
+export const symbolsGroup = {
```
Comment (mosheKerbel) [inline stories/symbolsComponentsMapping/sketchSymbols.js]: i don't see why we need to mention in the file name that it's from Sketch - `sketchSymbols`, i'd call it just `symbols.js`
but if you do think it's important, then consider adding a comment saying something like `Reflecting the symbol groups as described in UX Sketch`
also for the next colleciotn

File: stories/Introduction/Cheatsheet/sharedComponents/utils.js [hunk 23]
```diff
@@ -0,0 +1,24 @@
+import {
+  getComponentUrl,
+  getSymbolUrl,
+  componentsWithoutDocumentation,
+} from '../../../symbolsComponentsMapping/storybookMapping';
+
+export const linkedSymbolName = ({ groupSymbol, symbol }) => ({
```
Comment (mosheKerbel) [inline stories/Introduction/Cheatsheet/sharedComponents/utils.js]: if these functions are considered utils i would add a verb before, like `create`
`createLinkedSymbolName`

File: stories/symbolsComponentsMapping/storybookMapping.js (hunk 20 above)
Comment (TalZa1810) [reply stories/symbolsComponentsMapping/storybookMapping.js]: Some of the links are not in the same format as they are suppose to be- `groupSymbol` & `symbol`. This mapping gives the `actual` url to prevent broken links. Therefore the `actual` prefix is needed here (and in the `symbol variable`).
